.github/workflows/check-mergability.yaml

```diff
@@ -0,0 +1,38 @@
+# Checks for merge conflicts when a pull request is opened or updated
+name: Check mergability onto main
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  merge_check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # This is important to fetch all history
```

Comment:
Is it really? Why?

Reply:
oops you're right

Reply:
`fetch-depth: 0` is to get everything https://github.com/actions/checkout.  We usually avoid using that but I guess this is needed here to fetch the main branch?  Just FYI, I have been following this change https://github.com/actions/checkout/pull/579, which proposes a way to fetch tags without a full checkout